File: temp/fix_nested_quotes.py
#!/usr/bin/env python3
import os

base = r"D:\新课开发\数字化转型\4.价值密度战略设计：从交付产品到交付结果\授课PPT\slides"

# Fix slide-30 line 122
path = os.path.join(base, "slide-30.js")
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()
content = content.replace(
    "'核心区别：产品思维问\"做了什么'，结果思维问'解决了什么\"'",
    '"核心区别：产品思维问\\"做了什么\\"，结果思维问\\"解决了什么\\""'
)
with open(path, 'w', encoding='utf-8') as f:
    f.write(content)
print("Fixed slide-30")

# Fix slide-32 line 186
path = os.path.join(base, "slide-32.js")
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()
content = content.replace(
    "'从\"功能交付'到'价值交付\"'",
    '"从\\"功能交付\\"到\\"价值交付\\""'
)
with open(path, 'w', encoding='utf-8') as f:
    f.write(content)
print("Fixed slide-32")

# Fix slide-137 line 163
path = os.path.join(base, "slide-137.js")
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()
content = content.replace(
    "'核心转变：从\"以人为主'到'人机协同\"，智能体承担重复性任务，人类专注创造性决策'",
    '"核心转变：从\\"以人为主\\"到\\"人机协同\\"，智能体承担重复性任务，人类专注创造性决策"'
)
with open(path, 'w', encoding='utf-8') as f:
    f.write(content)
print("Fixed slide-137")

File: temp/fix-nested-quotes.js
const fs = require('fs');
const path = require('path');

const slidesDir = 'D:/新课开发/党业融合/经营者讲党课/完整课程包/003-授课PPT/slides';
const files = fs.readdirSync(slidesDir).filter(f => f.match(/^slide-\d+\.js$/)).sort();

files.forEach(file => {
  const filePath = path.join(slidesDir, file);
  let content = fs.readFileSync(filePath, 'utf8');
  
  // Fix nested quotes that broke after our simple replacement
  // Pattern: text followed by "word" where the inner " needs to become '
  // We need to replace "text" with \"text\" when inside a JS string
  
  // Simple approach: replace all remaining Chinese brackets first
  // Then escape any remaining double quotes that are inside strings
  
  // More targeted: find all addText calls and fix the content
  // Replace "..." inner quotes with \"...\" or '...'
  
  // Actually the safest approach: replace any isolated " that appears mid-string
  // by checking if it's between two other "
  
  // Let's use a different strategy - replace " that is surrounded by word chars with \"
  // This handles cases like: "不是"经验萃取转化"类" -> "不是\"经验萃取转化\"类"
  
  // But this is complex. Let's instead replace inner Chinese quotes with escaped version
  // before we replace all brackets.
  
  // Simpler: just replace the pattern where " is followed by Chinese chars and preceded by Chinese chars
  content = content.replace(/(?<=[一-龥])"(?=[一-龥])/g, '\\"');
  
  fs.writeFileSync(filePath, content);
});

console.log('Done - fixed nested quotes');
